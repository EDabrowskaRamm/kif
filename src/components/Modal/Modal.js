import React from 'react'

import {
  ModalOverlay,
  ModalRoot
} from './Modal.shards'

import { CloseButton, Button } from '../Button/Button'

const Modal = ({ modalActive, setModalActive }) => {
  const closeModal = () => setModalActive(false)
  const handleConfirm = () => console.log('confirmed')

  return (
    <ModalOverlay>
      <ModalRoot>
        <CloseButton handleClick={closeModal} />
        <h4>Podaj kod PIN</h4>
        <p>Na podany numer telefonu wysłaliśmy SMS z kodem PIN</p>
        <Button
          handleClick={handleConfirm}
          text='Potwierdź'
          active={false}
        />
        <p>Nie otrzymałeś kodu? <span>Wyślij ponownie</span></p>
      </ModalRoot>
    </ModalOverlay>
  )
}

export default Modal