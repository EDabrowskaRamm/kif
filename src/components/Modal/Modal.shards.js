import styled from '@emotion/styled'

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  background: ${({ theme }) => theme.colors.background.overlay};
`

export const ModalRoot = styled.div`
  position: relative;
  padding: 76px 20px 70px;
  background: ${({ theme }) => theme.colors.background.primary};
  border-radius: ${({ theme }) => theme.commons.borderRadius.modal};
  text-align: center;
  width: 90%;

  @media (min-width: 992px) {
    width: 100%;
    max-width: 800px;
    padding-top: 100px;
  }
`

export const ImgWrapper = styled.div`
  position: absolute;
  top: -40px;
  left: 50%;
  transform: translateX(-50%);
  display: flex;
  align-items: center;
  justify-content: center;
  width: 80px;
  height: 80px;
  border-radius: 50%;
  background-color: ${({ theme }) => theme.colors.background.accent};

  @media (min-width: 992px) {
    width: 100px;
    height: 100px;
  }
`

export const Title = styled.h4`
  font-style: normal;
  font-weight: bold;
  font-size: 2.2rem;
  line-height: 2.2rem;
  color: ${({ theme }) => theme.colors.text.title};
  margin: 12px 0;

  @media (min-width: 992px) {
    font-size: 2.4rem;
    line-height: 2.4rem;
    margin: 20px 0;
  }
`

export const Description = styled.p`
  font-style: normal;
  font-weight: normal;
  font-size: 1.4rem;
  line-height: 2.2rem;

  @media (min-width: 992px) {
    font-size: 1.6rem;
    line-height: 1.6rem;
  }
`

export const Note = styled.p`
  font-style: normal;
  font-weight: normal;
  font-size: 1.4rem;
  line-height: 1.6rem;

  @media (min-width: 992px) {
    font-size: 1.6rem;
  }
`

export const Marked = styled.span`
  color: ${({ theme }) => theme.colors.text.error};
  cursor: pointer;
`