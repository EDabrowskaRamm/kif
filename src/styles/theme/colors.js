const _white = '#FFF'
const _colorGrey47 = '#707181'

const _tuna = '#30313D'
const _stormGray = _colorGrey47
const _snuff = '#DBDCEE'
const _luckyPoint = '#211965'
const _bittersweet = '#FF7165'

export const background = {
  primary: _white,
  overlay: _tuna,
}

export const text = {
  primary: _stormGray,
  secondary: _white,
  title: _tuna,
  error: _bittersweet,
}

export const border = {
  input: _snuff,
  error: _bittersweet,
}

export const buttonActive = {
  text: _white,
  background: _bittersweet
}

export const buttonDisabled = {
  text: _luckyPoint,
  background: 'rgba(141, 142, 161, 0.1)'
}