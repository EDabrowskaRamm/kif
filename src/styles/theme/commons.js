import { border } from './colors'

export const borders = {
  input: `1px solid ${border.input}`,
  error: `1px solid ${border.error}`,
}

export const borderRadius = {
  button: '4px',
  input: '10px',
  modal: '16px'
}