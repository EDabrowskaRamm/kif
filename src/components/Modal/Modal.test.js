import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'

import Modal from './Modal'

import theme from '../../styles/theme/theme'

expect.addSnapshotSerializer(serializer)

describe('renders Modal', () => {
  it('Snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Modal setModalActive={() => { }} />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
