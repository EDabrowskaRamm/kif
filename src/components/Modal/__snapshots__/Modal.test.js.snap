// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Modal Snapshot 1`] = `
.emotion-9 {
  position: fixed;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  background: #30313D;
}

.emotion-8 {
  position: relative;
  padding: 76px 20px 70px;
  background: #FFF;
  border-radius: 16px;
  text-align: center;
  width: 90%;
}

@media (min-width:992px) {
  .emotion-8 {
    width: 100%;
    max-width: 800px;
    padding-top: 100px;
  }
}

.emotion-0 {
  position: absolute;
  top: -40px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 80px;
  height: 80px;
  border-radius: 50%;
  background-color: #FF7165;
}

@media (min-width:992px) {
  .emotion-0 {
    width: 100px;
    height: 100px;
  }
}

.emotion-1 {
  position: absolute;
  top: 20px;
  right: 20px;
  border-radius: 10px;
  border: 1px solid #DBDCEE;
  background: transparent;
  color: #30313D;
  padding: 10.5px 16.6px;
  font-size: 2.2rem;
  outline: none;
  cursor: pointer;
}

.emotion-2 {
  font-style: normal;
  font-weight: bold;
  font-size: 2.2rem;
  line-height: 2.2rem;
  color: #30313D;
  margin: 12px 0;
}

@media (min-width:992px) {
  .emotion-2 {
    font-size: 2.4rem;
    line-height: 2.4rem;
    margin: 20px 0;
  }
}

.emotion-3 {
  font-style: normal;
  font-weight: normal;
  font-size: 1.4rem;
  line-height: 2.2rem;
}

@media (min-width:992px) {
  .emotion-3 {
    font-size: 1.6rem;
    line-height: 1.6rem;
  }
}

.emotion-5 {
  position: relative;
  font-style: normal;
  font-weight: bold;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: 40px auto 20px;
  max-width: 290px;
}

.emotion-4 {
  border-radius: 4px;
  background-color: #FF7165;
  color: #FFF;
  padding: 16px 36px;
  outline: none;
  cursor: pointer;
  font-style: normal;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 1.6rem;
}

.emotion-4:disabled {
  color: #211965;
  background: rgba(141,142,161,0.1);
  cursor: default;
}

@media (min-width:992px) {
  .emotion-4 {
    font-size: 1.6rem;
  }
}

.emotion-7 {
  font-style: normal;
  font-weight: normal;
  font-size: 1.4rem;
  line-height: 1.6rem;
}

@media (min-width:992px) {
  .emotion-7 {
    font-size: 1.6rem;
  }
}

.emotion-6 {
  color: #FF7165;
  cursor: pointer;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-8"
  >
    <div
      className="emotion-0"
    >
      <img
        alt="lock"
        src="lock.png"
      />
    </div>
    <button
      className="emotion-1"
      onClick={[Function]}
      type="button"
    >
      ×
    </button>
    <h4
      className="emotion-2"
    >
      Podaj kod PIN
    </h4>
    <p
      className="emotion-3"
    >
      Na podany numer telefonu wysłaliśmy SMS z kodem PIN
    </p>
    <div
      className="emotion-5"
    >
      <div
        className="pincode-input-container"
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-evenly",
            "marginBottom": "40px",
          }
        }
      >
        <input
          aria-label=""
          autoComplete="off"
          className="pincode-input-text"
          maxLength="1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          pattern="[0-9]*"
          placeholder=""
          style={
            Object {
              "alignItems": "center",
              "background": "transparent",
              "border": "1px solid #DBDCEE",
              "borderRadius": "10px",
              "display": "flex",
              "height": "68px",
              "justifyContent": "center",
              "margin": "0 2px",
              "padding": "22px 18px",
              "textAlign": "center",
              "width": "50px",
            }
          }
          type="tel"
          value=""
        />
        <input
          aria-label=""
          autoComplete="off"
          className="pincode-input-text"
          maxLength="1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          pattern="[0-9]*"
          placeholder=""
          style={
            Object {
              "alignItems": "center",
              "background": "transparent",
              "border": "1px solid #DBDCEE",
              "borderRadius": "10px",
              "display": "flex",
              "height": "68px",
              "justifyContent": "center",
              "margin": "0 2px",
              "padding": "22px 18px",
              "textAlign": "center",
              "width": "50px",
            }
          }
          type="tel"
          value=""
        />
        <input
          aria-label=""
          autoComplete="off"
          className="pincode-input-text"
          maxLength="1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          pattern="[0-9]*"
          placeholder=""
          style={
            Object {
              "alignItems": "center",
              "background": "transparent",
              "border": "1px solid #DBDCEE",
              "borderRadius": "10px",
              "display": "flex",
              "height": "68px",
              "justifyContent": "center",
              "margin": "0 2px",
              "padding": "22px 18px",
              "textAlign": "center",
              "width": "50px",
            }
          }
          type="tel"
          value=""
        />
        <input
          aria-label=""
          autoComplete="off"
          className="pincode-input-text"
          maxLength="1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          pattern="[0-9]*"
          placeholder=""
          style={
            Object {
              "alignItems": "center",
              "background": "transparent",
              "border": "1px solid #DBDCEE",
              "borderRadius": "10px",
              "display": "flex",
              "height": "68px",
              "justifyContent": "center",
              "margin": "0 2px",
              "padding": "22px 18px",
              "textAlign": "center",
              "width": "50px",
            }
          }
          type="tel"
          value=""
        />
      </div>
      <button
        className="emotion-4"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Potwierdź
      </button>
    </div>
    <p
      className="emotion-7"
    >
      Nie otrzymałeś kodu?
      <span
        className="emotion-6"
        onClick={[Function]}
      >
        Wyślij ponownie
      </span>
    </p>
  </div>
</div>
`;
